server/libs/domain/src/metadata/metadata.service.ts

```diff
@@ -1,77 +1,54 @@
-import { AssetType } from '@app/infra/entities';
-import { Inject, Logger } from '@nestjs/common';
+import { Inject } from '@nestjs/common';
 import { constants } from 'fs/promises';
-import { AssetCore, IAssetRepository, WithoutProperty, WithProperty } from '../asset';
+import { IAssetRepository, WithoutProperty, WithProperty } from '../asset';
 import { usePagination } from '../domain.util';
-import { IAssetJob, IBaseJob, IJobRepository, JobName, JOBS_ASSET_PAGINATION_SIZE } from '../job';
+import { IBaseJob, IEntityJob, IJobRepository, JobName, JOBS_ASSET_PAGINATION_SIZE } from '../job';
 import { IStorageRepository } from '../storage';
 
 export class MetadataService {
-  private logger = new Logger(MetadataService.name);
-  private assetCore: AssetCore;
-
   constructor(
     @Inject(IAssetRepository) private assetRepository: IAssetRepository,
     @Inject(IJobRepository) private jobRepository: IJobRepository,
     @Inject(IStorageRepository) private storageRepository: IStorageRepository,
-  ) {
-    this.assetCore = new AssetCore(assetRepository, jobRepository);
-  }
+  ) {}
 
   async handleQueueSidecar(job: IBaseJob) {
-    try {
-      const { force } = job;
-      const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) => {
-        return force
-          ? this.assetRepository.getWith(pagination, WithProperty.SIDECAR)
-          : this.assetRepository.getWithout(pagination, WithoutProperty.SIDECAR);
-      });
-
-      for await (const assets of assetPagination) {
-        for (const asset of assets) {
-          const name = force ? JobName.SIDECAR_SYNC : JobName.SIDECAR_DISCOVERY;
-          await this.jobRepository.queue({ name, data: { asset } });
-        }
+    const { force } = job;
+    const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) => {
+      return force
+        ? this.assetRepository.getWith(pagination, WithProperty.SIDECAR)
+        : this.assetRepository.getWithout(pagination, WithoutProperty.SIDECAR);
+    });
+
+    for await (const assets of assetPagination) {
+      for (const asset of assets) {
+        const name = force ? JobName.SIDECAR_SYNC : JobName.SIDECAR_DISCOVERY;
+        await this.jobRepository.queue({ name, data: { id: asset.id } });
       }
-    } catch (error: any) {
-      this.logger.error(`Unable to queue sidecar scanning`, error?.stack);
     }
+
+    return true;
   }
 
-  async handleSidecarSync(job: IAssetJob) {
-    const { asset } = job;
-    if (!asset.isVisible) {
-      return;
-    }
+  async handleSidecarSync() {
+    // TODO: optimize to only queue assets with recent xmp changes
+    return true;
```

Comment:
That means sidecar sync (temporarily) doesn't work, right?

Reply:
It does work, the job doesn't do anything though. The onDone queues the next job though, which is what you are probably looking for.